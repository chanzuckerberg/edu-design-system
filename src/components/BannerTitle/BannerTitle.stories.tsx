import { Story, Meta } from '@storybook/react';
import React from 'react';

import { BannerTitle, Props } from './BannerTitle';

export default {
  title: 'Molecules/Messaging/BannerTitle',
  component: BannerTitle,
} as Meta;

const Template: Story<Props> = (args) => <BannerTitle {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Banner Title',
};
