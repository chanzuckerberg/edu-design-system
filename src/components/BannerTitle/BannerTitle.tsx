import React, { ReactNode } from 'react';
import Heading, { HeadingElement } from '../Heading';

export interface Props {
  /**
   * The element this component will render as
   */
  as?: HeadingElement;
  /**
   * CSS class names that can be appended to the component.
   */
  className?: string;
  /**
   * Children content
   */
  children: ReactNode;
}

/**
 * Used for the title text in the Banner.
 */
export const BannerTitle = ({ as = 'h3', children, ...other }: Props) => {
  return (
    <Heading as={as} size="h5" {...other}>
      {children}
    </Heading>
  );
};
