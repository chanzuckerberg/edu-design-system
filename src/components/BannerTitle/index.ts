export { BannerTitle as default } from './BannerTitle';
