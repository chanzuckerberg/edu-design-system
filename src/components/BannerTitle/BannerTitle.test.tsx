import { generateSnapshots } from '@chanzuckerberg/story-utils';
import * as stories from './BannerTitle.stories';

describe('<BannerTitle />', () => {
  generateSnapshots(stories);
});
