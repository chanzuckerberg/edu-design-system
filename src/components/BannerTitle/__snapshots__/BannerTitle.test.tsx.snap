// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BannerTitle /> Default story renders snapshot 1`] = `
<h3
  class="heading heading--size-h5"
>
  Banner Title
</h3>
`;
